tonic/src/extensions.rs

```diff
@@ -71,3 +71,12 @@ impl fmt::Debug for Extensions {
         f.debug_struct("Extensions").finish()
     }
 }
+
+/// A gRPC Method info extension.
+#[derive(Debug)]
+pub struct GrpcMethod<'a> {
+    /// gRPC service name
+    pub service: &'a str,
```

Comment:
These should both be moved to `'static` and be made private. We should then expose `service() -> &str` and `method() -> &str` to avoid exposing that 1) these are 'static (incase we want to change that in the future) and 2) in case we want to add new fields we can do that and 3) we remove the `'a` lifetime on the struct. Since, with extensions https://docs.rs/http/latest/http/struct.Extensions.html#method.insert always requires `'static` there is no point to have a lifetime.